manifests/resource/geo.pp

```diff
@@ -63,6 +63,10 @@
   Optional[Boolean] $proxy_recursive  = undef
 ) {
 
+  if ! defined(Class['nginx']) {
```

Comment:
I think this is fine and probably goes with the style of the module as a whole... that said, I think some people would say that 'unless' is more idiomatic, and probably lines up better with https://docs.puppet.com/puppet/4.9/lang_conditional.html ? I like 'if not' better than 'if !' but IIRC, it may not work in the DSL?